### ISSUE TITLE
Callable subtyping bug w/ positional-only params & unpacked *args

### ISSUE THREAD

**Issue by yangdanny97** [microsoft/pyright#9974]
If the annotation for *args is an unpacked concrete tuple (or resolves to some concrete tuple), I think it should be equivalent to required positional-only parameters.

```python
from typing import Protocol
class P1(Protocol):
    def __call__(self, *args: *tuple[int, int]): ...
class P2(Protocol):
    def __call__(self, x: int, y: int, /): ...
class P3(Protocol):
    def __call__(self, x: int, /, *args: *tuple[int]): ...
class P4(Protocol):
    def __call__(self, x: int, y: int = 2, /): ...

def test(p1: P1, p2: P2, p3: P3, p4: P4):
    x1: P1 = p2
    x2: P1 = p3
    x3: P1 = p4
```

Pyright gives an unexpected error for the 3 assignments inside `test()`, but mypy thinks they're all OK.

**Comment by erictraut** [comment]
Thanks for the bug report and the clear code sample.

This will be addressed in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.395.
